### ISSUE TITLE
[Training] Can I use the C++ interface of OnnxRuntime to train a model in Win32?

### ISSUE THREAD

**Issue by w1005444804** [microsoft/onnxruntime#20325]
### Describe the issue

I just want to use the C++interface to fine tune a onnx model. int the win32 system, however test with win-x64 firstly.

### To reproduce

float loss = *(session_cache->training_session.TrainStep(user_inputs).front().GetTensorMutableData<float>());
collapsed here！

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

onnxruntime-training-win-x64-1.15.1

### PyTorch Version

none

### Execution Provider

Default CPU

**Comment by w1005444804** [comment]
Can you provide a C++ example for training lenet.onnx
